pkg/http/listener.go

```diff
@@ -179,24 +179,8 @@ func (listener *httpListener) start() {
 			bufconn = newBufConn(tlsConn, listener.readTimeout, listener.writeTimeout,
 				listener.updateBytesReadFunc, listener.updateBytesWrittenFunc)
 
-			// Peek bytes of maximum length of all HTTP methods.
-			data, err := bufconn.Peek(methodMaxLen)
-			if err != nil {
-				if listener.errorLogFunc != nil {
-					listener.errorLogFunc(err,
-						"Error in reading from new TLS connection %s at server %s",
-						bufconn.RemoteAddr(), bufconn.LocalAddr())
-				}
-				bufconn.Close()
-				return
-			}
-
-			// Return bufconn if read data is a valid HTTP method.
-			tokens := strings.SplitN(string(data), " ", 2)
-			if isHTTPMethod(tokens[0]) {
-				send(acceptResult{bufconn, nil}, doneCh)
-				return
-			}
```

Comment:
This would be incorrect @vadmeste Peek is fine but we don't need to print for `io.EOF` that's all. Read the handleConn above..

```
  // Peek could fail legitimately when clients abruptly close
  // connection. E.g. Chrome browser opens connections speculatively to
  // speed up loading of a web page. Peek may also fail due to network
  // saturation on a transport with read timeout set. All other kind of
  // errors should be logged for further investigation. 
  if !ignoreErr(err) {
```

Reply:
We don't need peeking at this stage @harshavardhana, peeking is only to differentiate between TLS & plain HTTP, but here we are sure that this is a TLS connection because the handshake was already successful. Golang HTTP can handle HTTP errors already and we don't need to add our own code for that.

Reply:
No it is on purpose @vadmeste we are also validating if the HTTP method sent is correct as well.

Reply:
> No it is on purpose @vadmeste we are also validating if the HTTP method sent is correct as well.

Won't golang http also validate that ?

Reply:
We are not worried about that @vadmeste `pkg/http` is a lower level implementation with our own requirements.  It is because when you manage a `conn`  for HTTP you are supposed to handle everything related to HTTP at this level. Passing the issue to Go and expect it to fail in unexpected ways would be incosistent.  

`pkg/http` is not a thin layer implementation it is purposefully built to handle these conditions.

Reply:
> Passing the issue to Go and expect it to fail in unexpected ways would be incosistent.

Are you saying that pkg/http should validate all the http protocol ?

Reply:
I mean the Conn validation. Not HTTP parsing. Simple fix is what I mentioned above. 

Removing method validation is incorrect.